flow/diff_context_unittests.cc

```diff
@@ -0,0 +1,293 @@
+#include "diff_context.h"
+#include "flutter/flow/layers/container_layer.h"
+#include "flutter/flow/layers/picture_layer.h"
+#include "flutter/flow/testing/diff_context_test.h"
+#include "gtest/gtest.h"
+
+namespace flutter {
+namespace testing {
+
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+TEST_F(DiffContextTest, SimplePicture) {
+  auto picture = CreatePicture(SkRect::MakeXYWH(10, 10, 50, 50), 1);
```

Comment:
I would expect the file `diff_context_unittests.cc` to have tests for just the DiffContext class itself. Most of these tests look like testing whether PictureLayer will correctly determine whether it has changed or not. I would expect tests like that to go into `picture_layer_unittests.cc`, for instance.

New layers, deleted layers, moved layers - would probably belong in `container_layer_unittests.cc`?

Reply:
I'm not quite sure how much of DiffContext functionality can be tested without using Layers for diffing. There are also some tests that use both ContainerLayer and PictureLayer, not sure where that would go. I guess for ones specifically testing ContainerLayer (add/remove) we could use MockLayer for children, and then tests for other layers would just use ContainerLayer.

But I also have plans for more complex tests (i.e. mixing transforms / clips / etc) so where would this go?

Reply:
There may be no need for diff_context_unittests if there is nothing to test about the DiffContext itself. I could see a few tests, though, related to calling AddDamage a few times and then making sure the damage output contains all of the source rects, etc.

There will likely be some containers involved in a few of the tests, the question is not whether their presence makes it a container layer test, but whether the test is targeted at verifying something that the container layer does, or something that the other (i.e. picture layer) does.